pkg/resolution/resource/name.go

```diff
@@ -69,3 +79,25 @@ func GenerateDeterministicName(prefix, base string, params v1.Params) (string, e
 	}
 	return fmt.Sprintf("%s-%x", prefix, hasher.Sum(nil)), nil
 }
+
+func GenerateErrorLogString(resolverType string, params v1.Params) string {
+	paramString := fmt.Sprintf("resolver type %s\n", resolverType)
+	for _, p := range params {
+		if p.Name == ParamName {
+			name := p.Value.StringVal
+			if p.Value.Type != v1.ParamTypeString {
+				asJSON, err := p.Value.MarshalJSON()
+				if err != nil {
+					paramString += fmt.Sprintf("name could not be marshalled: %s\n", err.Error())
+					continue
+				}
+				name = string(asJSON)
+			}
+			paramString += fmt.Sprintf("name = %s\n", name)
+		}
+		if p.Name == ParamURL {
+			paramString += fmt.Sprintf("url = %s\n", p.Value.StringVal)
+		}
+	}
+	return paramString
+}
```

Comment:
Can we add tests for this exported function? And also some doc string? Thanks!

Reply:
updated - thanks !

Reply:
Thanks!